images/bridge2/assistant/assistant.go

```diff
@@ -0,0 +1,144 @@
+package assistant
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/ajbouh/substrate/images/bridge2/tracks"
+	"github.com/ajbouh/substrate/images/bridge2/transcribe"
+)
+
+var recordAssistant = tracks.EventRecorder[*AssistantEvent]("assistant")
+
+type AssistantEvent struct {
+	SourceEvent tracks.ID
+	Name        string
+	Prompt      string
+	Response    string
+	Error       string
+}
+
+type Agent struct {
+	Assistants map[string]string
+}
+
+func (a *Agent) HandleEvent(annot tracks.Event) {
+	if annot.Type != "transcription" {
+		return
+	}
+	in := annot.Data.(*transcribe.Transcription)
+
+	if len(in.Segments) == 0 || len(in.Segments[0].Words) == 0 {
+		log.Printf("assistant: transcription had no segments or words")
+		return
+	}
+	text := strings.TrimSpace(in.Text())
+	names := a.matchAssistants(text)
+	log.Printf("assistant: matched %v for: %s", names, text)
+	for _, name := range names {
+		go a.respond(annot, name, text)
+	}
+}
+
+func (a *Agent) matchAssistants(text string) []string {
+	var matched []string
+	text = strings.ToLower(text)
+	for name := range a.Assistants {
+		if strings.Contains(text, name) {
+			matched = append(matched, name)
+		}
+	}
+	return matched
+}
+
+func (a *Agent) respond(annot tracks.Event, name, prompt string) {
+	out := AssistantEvent{
+		SourceEvent: annot.ID,
+		Name:        name,
+		Prompt:      prompt,
```

Comment:
i think we should call the thing the user asks for something other than "prompt", since it gets combined with other things to form the actual "prompt"

Reply:
Ok, how about "input"?

Reply:
yes, that sounds right